### ISSUE TITLE
“No such file or directory” when trying to use holyHammer

### ISSUE THREAD

**Issue by marioxcc** [HOL-Theorem-Prover/HOL#566]
When I try to use holyHammer with E 2.0 in HOL4 commit 37bf2ea80dd3e83d:
```
> g ‘x = x’;
<<HOL message: inventing new type variable names: 'a>>
val it =
   Proof manager status: 1 proof.
   1. Incomplete goalstack:
        Initial goal:
        
        x = x
        
        
   : proofs
> e(hh);
OK..
SysErr ("No such file or directory", SOME ENOENT)Exception- SysErr ("No such file or directory", SOME ENOENT) raised
```

Running within “strace” gives the following relevant fragment:
```
open("/home/marioxcc/hacking/hol_git/src/tactictoe/gen_code/scratch_hide_out", O_WRONLY|O_CREAT|O_TRUNC, 0666) = -1 ENOENT (No such file or directory)
```
The directory “gen_code” does not exist. A workaround is to manually create the directory “gen_code”, then the call to `hh` works as expected in this trivial case:
```
> e(hh);
OK..
Loading 2995 theorems 
Proof found by eprover:
  metisTools.METIS_TAC []
Minimized proof:
  METIS_TAC []
metis: val it =
   Initial goal proved.
   ⊢ x = x: proof
```

P.S: Programs should not use their build directory to create scratch files, they should create them under the directory pointed by the environment variable `TMP` or perhaps `XDG_CACHE_HOME`.

**Comment by barakeel** [comment]
I have added the missing directory. I am not sure about how to guarantee that no other program will modify the file if I write the file in /tmp.

**Comment by mn200** [comment]
I assume this is now fixed (see 2e21646)
